docs/okeps/okep-5193-user-defined-networks.md

```diff
@@ -0,0 +1,1468 @@
+effect on its functionality.
+
+##### Egress Service
+
+Egress service is namespace scoped and its primary function is to SNAT egress packets to a load balancer IP. As
+previously mentioned, the feature works the same in shared and local gateway mode, by leveraging the local gateway mode
+path. Therefore, its design will be covered in the Local Gateway Mode section of the Design Details.
+
+##### Multiple External Gateways (MEG)
+
+There will be no support for MEG or pod direct ingress on any network other than the primary, cluster default network.
+This support may be enhanced later by extending VRFs/networks outside the cluster.
+
+#### Local Gateway Mode
+
+With local gateway mode, egress/ingress traffic uses the kernel’s networking stack as a next hop. OVN-Kubernetes
+leverages an interface named “ovn-k8s-mp0” in order to facilitate sending traffic to and receiving traffic from the
+host. For egress traffic, the host routing table decides where to send the egress packet, and then the source IP is
+masqueraded to the node IP of the egress interface. For ingress traffic, the host routing table steers packets destined
+for pods via ovn-k8s-mp0 and SNAT’s the packet to the interface address.
+
+For multiple networks to use local gateway mode, some changes are necessary. The ovn-k8s-mp0 port is a logical port in
+the OVN topology tied to the cluster default network. There will need to be multiple ovn-k8s-mp0 ports created, one per
+network. Additionally, all of these ports cannot reside in the default VRF of the host network. Doing so would result in
+an inability to have overlapping subnets, as well as the host VRF would be capable of routing packets between namespace
+networks, which is undesirable. Therefore, each ovn-k8s-mp0-&lt;network> interface must be placed in its own VRF:
+
+![Local GW Node Setup](../images/local-gw-node-setup-vrfs.svg)
+
+The VRFs will clone the default routing table, excluding routes that are created by OVN-Kubernetes for its networks.
+This is similar to the methodology in place today for supporting
+[Egress IP with multiple NICs](https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/docs/features/cluster-egress-controls/egress-ip.md#egressip-ip-is-assigned-to-a-secondary-host-interface).
+
+##### Pod Egress
+
+Similar to the predicament outlined in Shared Gateway mode, we need to solve the improbable case where two networks have
+the same address space, and pods with the same IP/ports are trying to talk externally to the same server. In this case,
+OVN-Kubernetes will reserve an extra IP from the masquerade subnet per network. This masquerade IP will be used to SNAT
+egress packets from pods leaving via mp0. The SNAT will be performed by ovn_cluster_router for layer 3 networks and
+the gateway router (GR) for layer 2 networks using configuration like:
+
+```
+ovn-nbctl --gateway-port=rtos-ovn-worker lr-nat-add DR snat 10.244.0.6 169.254.169.100
```

Comment:
I think we match on MAC address IIRC but design wise we don't care about the accurate details I assume

Reply:
i'll take a look